controllers/istio_controller.go

```diff
@@ -153,6 +153,11 @@ func (r *IstioReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
 
 	values := istio.Spec.GetValues()
 
+	if err := istio.Spec.SetValues(mergeValues(values, istio.Spec.GetRawValues())); err != nil {
```

Comment:
You don't need to (and probably shouldn't) call `SetValues()` here. Simply merge values and rawValues and pass them to `installHelmCharts()`. 

On the other hand, you do want to store the merged values into `Istio.status.appliedValues`.